tests/end_to_end/test_letter.py

```diff
@@ -1,62 +1,65 @@
 import pytest
 import os
 from lib import Assertions, Generators, Helper
+from lib.fixtures import *
 from notifications_python_client.notifications import NotificationsAPIClient
 
 
 @pytest.mark.e2e
 @pytest.mark.devtest
-@pytest.mark.nhsd_apim_authorization({"access": "application", "level": "level3"})
-def test_letter_end_to_end_internal_dev(nhsd_apim_proxy_url, nhsd_apim_auth_headers):
+def test_letter_end_to_end_internal_dev(nhsd_apim_proxy_url, bearer_token_internal_dev):
     """
     .. include:: ../../partials/happy_path/test_letter_end_to_end_internal_dev.rst
     """
     resp = Helper.send_single_message(
         nhsd_apim_proxy_url,
-        nhsd_apim_auth_headers,
-        Generators.generate_send_message_body("letter", "internal-dev")
+        {"Authorization": bearer_token_internal_dev},
+        Generators.generate_send_message_body("letter", "internal-dev"),
     )
 
     message_id = resp.json().get("data").get("id")
 
     Helper.poll_get_message(
         url=nhsd_apim_proxy_url,
-        auth=nhsd_apim_auth_headers,
+        auth={"Authorization": bearer_token_internal_dev},
         message_id=message_id,
         end_state="sending",
-        poll_time=595
+        poll_time=595,
     )
 
     notifications_client = NotificationsAPIClient(os.environ.get("GUKN_API_KEY"))
-    gov_uk_response = notifications_client.get_all_notifications("received").get("notifications")
+    gov_uk_response = notifications_client.get_all_notifications("received").get(
+        "notifications"
+    )
 
     Assertions.assert_letter_gov_uk(gov_uk_response, message_id)
 
 
 @pytest.mark.e2e
 @pytest.mark.uattest
-@pytest.mark.nhsd_apim_authorization({"access": "application", "level": "level3"})
-def test_letter_end_to_end_uat(nhsd_apim_proxy_url, nhsd_apim_auth_headers):
+def test_letter_end_to_end_uat(nhsd_apim_proxy_url, bearer_token_internal_dev):
     """
     .. include:: ../../partials/happy_path/test_letter_end_to_end_uat.rst
     """
     resp = Helper.send_single_message(
         nhsd_apim_proxy_url,
-        nhsd_apim_auth_headers,
-        Generators.generate_send_message_body("letter", "internal-qa")
+        {"Authorization": bearer_token_internal_dev},
+        Generators.generate_send_message_body("letter", "internal-qa"),
```

Comment:
Minor, but I realise this `bearer_token_internal_dev` is being used for all the non production tests would it not be better named `bearer_token_non_prod`?  I'm presuming it's this token that provides access to the comms_manager_local app in APIGEE

Reply:
In terms of what the fixture is called I'm not too fussy but this is liable to change again when we add a second apigee client. How would we differentiate between the two nonprod tokens that we might have? Is int prod or nonprod? By prod do we mean live environments or path to live environments and what is in our path to live?

`internal-dev` at least sticks to the convention we're using in `AuthenticationCache` whereby it calls `.generate_authentication('internal-dev')`, seems weirder to me that we'd be calling a fixture called `bearer_token_nonprod` which in turn gets its token by calling `authentication_cache.generate_authentication('internal-dev')`. At least this way it's consistent.

If you disagree I'm happy to rename it to what you want since it's not too deep a decision and it's just a search and replace job at this point. I just want us to be very firm on what we want because it's referenced in so many places that it's a pain if we keep going back and forth on this.

Reply:
No problem, upon realising `authentication_cache.generate_authentication('internal-dev')` had been called all along and you've just created the wrapper on top of that, we can leave it be for now